icu4c/source/tools/toolutil/pkg_genc.cpp

```diff
@@ -1254,11 +1257,17 @@ writeObjectCode(
     uprv_memset(&symbolNames, 0, sizeof(symbolNames));
 
     /* write the linker export directive */
-    uprv_strcpy(objHeader.linkerOptions, "-export:");
+    if (optWinDllExport) {
+        uprv_strcpy(objHeader.linkerOptions, "-export:");
+    }
     length=8;
```

Comment:
The length tracks the number of characters in objHeader.linkerOptions.

I think you need to set length=0 before your first "if", then length+=8/entryLength/6 after each strcpy.
Which then wants to collapse to something like
```
    if (optWinDllExport) {
        uprv_strcpy(objHeader.linkerOptions, "-export:");
        length=8;
        uprv_strcpy(objHeader.linkerOptions+length, entry);
        length+=entryLength;
        uprv_strcpy(objHeader.linkerOptions+length, ",data ");
        length+=6;
    } else {
        length=0;
    }
```
unless I am missing something

Reply:
Ok, I'll test it and update.

Reply:
Done.